client/packages/system/src/Patient/VaccinationCards/DetailView/DetailView.tsx

```diff
@@ -12,30 +11,63 @@ import {
   NothingHere,
   LocaleKey,
   useFormatDateTime,
+  useRowStyle,
+  useTheme,
+  StatusCell,
 } from '@openmsupply-client/common';
 import {
   usePatientVaccineCard,
   VaxCardData,
 } from '../../api/hooks/usePatientVaccineCard';
 
+const useStyleRowsByStatus = (rows: DoseRowData[]) => {
+  const { setRowStyles } = useRowStyle();
+  const theme = useTheme();
+
+  useEffect(() => {
+    if (!rows) return;
+
+    setRowStyles(
+      rows.filter(row => !!row.dateGiven).map(row => row.id),
+      {
+        backgroundColor: `${theme.palette.background.success} !important`,
+      }
+    );
+  }, [rows]);
+};
+
 export const VaccinationCardComponent: FC = () => {
   const t = useTranslation('dispensary');
   const { patientId = '', programEnrolmentId = '' } = useParams();
   const { setCustomBreadcrumbs } = useBreadcrumbs();
   const { localisedDate } = useFormatDateTime();
+  const theme = useTheme();
 
   const {
     query: { data, isLoading },
   } = usePatientVaccineCard(patientId, programEnrolmentId);
 
-  const tableData = buildTableData(data);
+  const tableData = useMemo(() => buildTableData(data), [data]);
```

Comment:
Need to memo-ise this to avoid infinite re-render. But we maybe able to remove this after the back-end is wired up.